pkg/controller/rds/dbinstance/setup.go

```
@@ -0,0 +1,190 @@
+	client svcsdkapi.RDSAPI
+}
+
+func preObserve(_ context.Context, cr *svcapitypes.DBInstance, obj *svcsdk.DescribeDBInstancesInput) error {
+	obj.DBInstanceIdentifier = aws.String(meta.GetExternalName(cr))
+	return nil
+}
+
+func (e *custom) preCreate(ctx context.Context, cr *svcapitypes.DBInstance, obj *svcsdk.CreateDBInstanceInput) error {
+	pw, _, err := rds.GetPassword(ctx, e.kube, cr.Spec.ForProvider.MasterUserPasswordSecretRef, cr.Spec.WriteConnectionSecretToReference)
+	if resource.IgnoreNotFound(err) != nil {
+		return errors.Wrap(err, "cannot get password from the given secret")
+	}
+	if pw == "" && cr.Spec.ForProvider.AutogeneratePassword {
+		pw, err = password.Generate()
+		if err != nil {
+			return errors.Wrap(err, "unable to generate a password")
+		}
+		if err := e.savePasswordSecret(ctx, cr, pw); err != nil {
+			return errors.Wrap(err, errSaveSecretFailed)
+		}
+	}
+	obj.MasterUserPassword = aws.String(pw)
+	obj.DBInstanceIdentifier = aws.String(meta.GetExternalName(cr))
+	obj.VpcSecurityGroupIds = make([]*string, len(cr.Spec.ForProvider.VPCSecurityGroupIDs))
+	for i, v := range cr.Spec.ForProvider.VPCSecurityGroupIDs {
+		obj.VpcSecurityGroupIds[i] = aws.String(v)
+	}
+	obj.DBSecurityGroups = make([]*string, len(cr.Spec.ForProvider.DBSecurityGroups))
+	for i, v := range cr.Spec.ForProvider.DBSecurityGroups {
+		obj.DBSecurityGroups[i] = aws.String(v)
+	}
+	return nil
+}
+
+func (e *custom) postCreate(ctx context.Context, cr *svcapitypes.DBInstance, _ *svcsdk.CreateDBInstanceOutput, _ managed.ExternalCreation, err error) (managed.ExternalCreation, error) {
+	if err != nil {
+		return managed.ExternalCreation{}, err
+	}
+	conn := managed.ConnectionDetails{
+		xpv1.ResourceCredentialsSecretUserKey: []byte(aws.StringValue(cr.Spec.ForProvider.MasterUsername)),
+	}
+	pw, _, err := rds.GetPassword(ctx, e.kube, cr.Spec.ForProvider.MasterUserPasswordSecretRef, cr.Spec.WriteConnectionSecretToReference)
+	if resource.IgnoreNotFound(err) != nil {
```

```suggestion
	if err != nil {
```

We shouldn't actually ignore this error since that's the only way user can provide a password.